### ISSUE TITLE
Doc generation for explanations of metrics

### ISSUE THREAD

**Issue by sooheelee** [broadinstitute/gatk#3947]
Continues discussion in https://github.com/broadinstitute/gatk/issues/3853.
We need to generate GATK forum docs for [Picard metrics](http://broadinstitute.github.io/picard/picard-metric-definitions.html). We will classify the group as a utility, like we do `Read Filters` and `Variant Annotations`. Given GATK also has some metrics collection tools, the group name should be generic e.g. `Metrics`.

## @cmnbroad @magicDGS, What do I need to for each metric to show up in a metric doc as a table (like below)? 
![screenshot 2017-12-12 11 12 29](https://user-images.githubusercontent.com/11543866/33894997-63927c1a-df2d-11e7-8fd2-9c74809b7773.png)

---

### A test branch is `shl_doc_picardmetrics` in the Picard repo. 
Test is with HelpConstants.java and AlignmentSummaryMetrics.java.
- There is already a Picard program group `Metrics` (that should go to the wayside with the new categories in #3853) that should really be used for explaining the metrics. Metrics collecting tools go under the DiagnosticsAndQCProgramGroup.java.
- Given GATK has a number of metrics collection tools, the name is generic and currently, a placeholder: 

![screenshot 2017-12-12 11 03 43](https://user-images.githubusercontent.com/11543866/33894915-28340904-df2d-11e7-8363-72aabc76a490.png)

---
### Only the summary shows up:

![screenshot 2017-12-12 11 03 57](https://user-images.githubusercontent.com/11543866/33894921-2d7cbfa0-df2d-11e7-9c33-2cc8eb7a1e49.png)

---
### Here are the corresponding code elements:

![screenshot 2017-12-12 11 04 51](https://user-images.githubusercontent.com/11543866/33894926-3134e938-df2d-11e7-9690-bb2cd3f5408f.png)

**Comment by sooheelee** [comment]
@magicDGS What do you mean by a custom template?

**Comment by magicDGS** [comment]
The barclay's `DocWorkUnitHandler.getTemplateName(DocWorkUnit)` method can select a  template (Freemarker template) for a concrete documented feature. My suggestion was the following:

* Annotate fields in metrics files with `@Argument`
* Create a template for Freemarker specific for metrics: arguments will be included in a table with field and description
* In the GATK's implementation of DocWorkUnitHandler.getTemplateName, select that template for documented features that belongs to the metrics group

What do you think about it, @cmnbroad? I have been thinking about using that approach with a `@DocumentedFeature` for my own project - that's why I chime in here...

**Comment by magicDGS** [comment]
You're right, @cmnbroad - the `@Argument` annotation is unsafe if it is included in an argument collection. Maybe a `@DocumentedField`  annotation in barclay will help, by populating its information and include it in the json. This will allow to document also constant fields specific to a tool (e.g., a default value for a BAM tag, explaining why it is the default). That will be nice for several use cases, including the one here...

**Comment by cmnbroad** [comment]
Depends on https://github.com/broadinstitute/barclay/issues/128.
